tests/models/test_task.py

```diff
@@ -63,14 +65,17 @@ def test_PointingReconstructionWithKappa():
         target_labels=["zenith", "azimuth"],
         loss_function=GaussianNLLLoss(),
     )
-    
 
-def main():
-    # print(getmembers(__file__, isfunction))
 
-    # Test task
-    # test_transform_prediction_and_target()
-    test_PointingReconstructionWithKappa()
+def is_function_local(object):
+    return isinstance(object, types.FunctionType) and object.__module__ == __name__
+
+
+def main():
+    # Test tasks in script
+    function_names = [name for name, _ in getmembers(sys.modules[__name__], predicate=is_function_local)]
```

Comment:
I have added a for loop that tests all (meaningful) functions in the script.
I thought it could help give helpful information when running the unit test script.
Thoughts?

Reply:
We're using `pytest` for our unit testing suite (see [here](https://github.com/graphnet-team/graphnet/blob/main/setup.py#L37) and [here](https://github.com/graphnet-team/graphnet/blob/main/.github/workflows/build-matrix.yml#L29-L33)), so you should be able to run all of the relevant functions (i.e., the ones named `test_*`) using:
```
$ pytest tests/models/test_task.py
```
I can update the [contributing guide](https://github.com/graphnet-team/graphnet/blob/main/CONTRIBUTING.md) to reflect this, if you think that would be helpful? :)

Reply:
I think it is important to be transparent about the functions used in the framework and that they live up to a predictable output. Moving this transparency more "in front" would solidify the validity of the analysis results.
Correct me if I am wrong but [this](https://codeclimate.com/github/graphnet-team/graphnet) shows how much of the code/functions are currently tested. if not we should have something that shows how many of the functions pass tests..
Taking it further, it could be a "check item" (I think?) that requires a function to be testable and have a test before it can be submitted as a valid PR. But that is more in your ballpark of doability.